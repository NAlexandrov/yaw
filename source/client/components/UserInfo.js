
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import Gravatar from 'react-gravatar';
import { Icon } from 'antd';

const User = styled.div`
	display: flex;
	align-items: center;
	font-size: 15px;
	color: #000;
`;

const UserInfo = ({ user }) => (
  <User>
    <Gravatar email={user.email} size={42} default='mm' />
    {user.name || user.login} &nbsp;
    <a href='/logout'>
      <Icon type='logout' style={{ fontSize: 16, color: '#08c' }} />
    </a>
  </User>
);

UserInfo.propTypes = {
  user: PropTypes.shape({
    login: PropTypes.string,
    name: PropTypes.string,
    email: PropTypes.string,
  }).isRequired,
};

export default UserInfo;
