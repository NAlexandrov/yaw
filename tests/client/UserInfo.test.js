import React from 'react';
import UserInfo from '../../source/client/components/UserInfo';

describe('<UserInfo />', () => {
  test('should render a button', () => {
    const wrapper = shallow(<UserInfo />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render an avatar', () => {
    const user = {
      login: 'admin',
    };

    const wrapper = shallow(<UserInfo user={user} />);
    expect(wrapper).toMatchSnapshot();
  });
});
