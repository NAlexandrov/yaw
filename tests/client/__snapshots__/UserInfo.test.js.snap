// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> should render a button 1`] = `
<Button>
  Войти
</Button>
`;

exports[`<UserInfo /> should render an avatar 1`] = `
<styled(Component)>
  <styled(Component)
    src="/assets/avatar.png"
  />
  admin
</styled(Component)>
`;
